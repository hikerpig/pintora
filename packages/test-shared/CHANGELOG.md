# Change Log

## 0.2.0-alpha.3

### Patch Changes

- bc16cae: [gantt] add support for axisInterval
- 46a1a9f: [gantt] add support for axisInterval

## 0.2.0-alpha.2

### Patch Changes

- 235cc0e: add @config and @param to gantt diagram; update tmLanguage;

## 0.2.0-alpha.1

### Minor Changes

- bd214c7: Add gantt diagram

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2-beta.0](https://github.com/hikerpig/pintora/compare/v0.1.1...v0.1.2-beta.0) (2022-02-13)

### Bug Fixes

- should support `\r` as end-of-line char and whitespace char ([f0696f4](https://github.com/hikerpig/pintora/commit/f0696f48dcb2e82b7368b8a2908dc938b6c3aea2))

# [0.1.0](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.27...v0.1.0) (2022-02-06)

**Note:** Version bump only for package @pintora/test-shared

# [0.1.0-alpha.24](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.23...v0.1.0-alpha.24) (2022-02-01)

### Features

- **pintora-diagrams:** [componentDiagram] we can have relationship from element to group ([9cc74cd](https://github.com/hikerpig/pintora/commit/9cc74cdc53013b6b63a13658ff719d414d04706f))

# [0.1.0-alpha.19](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.18...v0.1.0-alpha.19) (2022-01-22)

### Features

- **pintora-diagrams:** add mindmap support ([3062216](https://github.com/hikerpig/pintora/commit/306221660b34baad83d67435470d6aabc27b289f))

# [0.1.0-alpha.18](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.17...v0.1.0-alpha.18) (2022-01-16)

### Features

- **pintora-cli:** read cli args `--theme` ([a502c35](https://github.com/hikerpig/pintora/commit/a502c35cf1dbbad65c5847e2752e17711b8dccef))

# [0.1.0-alpha.11](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.10...v0.1.0-alpha.11) (2021-12-04)

### Features

- **pintora-diagrams:** Add basic activityDiagram [#3](https://github.com/hikerpig/pintora/issues/3) ([718a80e](https://github.com/hikerpig/pintora/commit/718a80e55d209d618fb855a463a151b880bf6fc3))

# [0.1.0-alpha.8](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.7...v0.1.0-alpha.8) (2021-08-29)

### Features

- [sequenceDiagram] add custom group color ([f93fe8a](https://github.com/hikerpig/pintora/commit/f93fe8a6fdb81af55eeb6f303f426c6150eeaf86))

# [0.1.0-alpha.4](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.3...v0.1.0-alpha.4) (2021-08-15)

### Bug Fixes

- **pintora-diagrams:** [componentDiagram] avoid interface node overlapping ([213cea0](https://github.com/hikerpig/pintora/commit/213cea090f40717390417e143366026d841f9808))

### Features

- **pintora-diagrams:** [erDiagram] add support for key coloumn like 'PK' ([64dad3d](https://github.com/hikerpig/pintora/commit/64dad3ddd8cefb912d7c83e14cf8a2dc0e64c5f8))

# [0.1.0-alpha.0](https://github.com/hikerpig/pintora/compare/v0.0.1...v0.1.0-alpha.0) (2021-08-01)

### Features

- Add componentDiagram ([c967f1c](https://github.com/hikerpig/pintora/commit/c967f1c9a969a813adac9a9967a06a024f04d73f))
