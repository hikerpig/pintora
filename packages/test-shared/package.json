{
  "name": "@pintora/test-shared",
  "version": "0.2.0-alpha.3",
  "description": "Shared testing utils for pintora",
  "keywords": [
    "pintora"
  ],
  "author": "hikerpig <hikerpigwinnie@gmail.com>",
  "homepage": "https://github.com/hikerpig/pintora#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "sideEffects": false,
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hikerpig/pintora.git"
  },
  "scripts": {
    "compile": "tsc",
    "watch": "tsc -w"
  },
  "bugs": {
    "url": "https://github.com/hikerpig/pintora/issues"
  },
  "gitHead": "92814f05b3df654b588fe35e3af275a60ea61296"
}
