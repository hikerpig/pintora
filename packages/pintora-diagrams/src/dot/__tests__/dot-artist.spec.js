import { diagramRegistry } from '@pintora/core'
import { EXAMPLES } from '@pintora/test-shared'
import { testDraw, prepareDiagramConfig, stripDrawResultForSnapshot } from '../../__tests__/test-util'
import { dotDiagram } from '../index'

describe('dot-artist', () => {
  beforeAll(() => {
    prepareDiagramConfig()
    diagramRegistry.registerDiagram('dotDiagram', dotDiagram)
  })

  it('match example snapshot', () => {
    expect(stripDrawResultForSnapshot(testDraw(EXAMPLES.dot.code))).toMatchSnapshot()
  })

  it('renders different arrow type', () => {
    const code = `
    dotDiagram
    digraph Test {
      bgcolor="lightyellow";
      label="Graph Label";

      a1 -> b1 [arrowhead="box"];
      n1 -> end [arrowhead="odot"];
      a2 -> end [arrowhead="open"];
    }
    `
    expect(stripDrawResultForSnapshot(testDraw(code))).toMatchSnapshot()
  })
})
