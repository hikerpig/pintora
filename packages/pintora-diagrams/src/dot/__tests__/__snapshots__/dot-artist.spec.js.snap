// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dot-artist match example snapshot 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "attrs": {
          "fill": "white",
          "stroke": "transparent",
        },
        "class": "dot__frame-bg",
        "type": "rect",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "fill": "white",
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "fontWeight": "normal",
              "text": "Sub",
              "textAlign": "center",
              "textBaseline": "hanging",
            },
            "type": "text",
          },
          {
            "attrs": {
              "fill": "orange",
              "lineDash": null,
              "lineWidth": 1,
              "radius": 2,
              "stroke": "#111",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "purple",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "fontWeight": "normal",
              "text": "a1",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "lineDash": null,
              "lineWidth": 1,
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "white",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "lineDash": null,
              "lineWidth": 1,
              "stroke": "blue",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "blue",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "lineDash": null,
              "lineWidth": 1,
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {},
        "mark": {
          "attrs": {},
          "children": [
            {
              "attrs": {
                "fill": "orange",
                "lineDash": null,
                "lineWidth": 1,
                "radius": 2,
                "stroke": "#111",
              },
              "type": "path",
            },
          ],
          "type": "group",
        },
        "type": "symbol",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "end",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "fill": "orange",
          "lineDash": null,
          "lineWidth": 1,
          "radius": 2,
          "stroke": "#111",
        },
        "type": "rect",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "b1",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "fill": "orange",
          "lineDash": null,
          "lineWidth": 1,
          "radius": 2,
          "stroke": "#111",
        },
        "type": "rect",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "c1",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "Dot Diagram Example",
          "textAlign": "center",
          "textBaseline": "hanging",
        },
        "type": "text",
      },
    ],
    "type": "group",
  },
}
`;

exports[`dot-artist renders different arrow type 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "attrs": {
          "fill": "lightyellow",
          "stroke": "transparent",
        },
        "class": "dot__frame-bg",
        "type": "rect",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "lineDash": null,
              "lineWidth": 1,
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "lineDash": null,
              "lineWidth": 1,
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "lightyellow",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "lineDash": null,
              "lineWidth": 1,
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#3b4044",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "lineDash": null,
          "lineWidth": 1,
          "radius": 2,
          "stroke": "#3b4044",
        },
        "type": "rect",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "a1",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "lineDash": null,
          "lineWidth": 1,
          "radius": 2,
          "stroke": "#3b4044",
        },
        "type": "rect",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "b1",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "lineDash": null,
          "lineWidth": 1,
          "radius": 2,
          "stroke": "#3b4044",
        },
        "type": "rect",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "n1",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "lineDash": null,
          "lineWidth": 1,
          "radius": 2,
          "stroke": "#3b4044",
        },
        "type": "rect",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "end",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "lineDash": null,
          "lineWidth": 1,
          "radius": 2,
          "stroke": "#3b4044",
        },
        "type": "rect",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "a2",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "text": "Graph Label",
          "textAlign": "center",
          "textBaseline": "hanging",
        },
        "type": "text",
      },
    ],
    "type": "group",
  },
}
`;

exports[`dot-artist renders different node type 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "attrs": {},
        "mark": {
          "attrs": {},
          "children": [
            {
              "attrs": {
                "lineDash": null,
                "lineWidth": 1,
                "radius": 2,
                "stroke": "#3b4044",
              },
              "type": "ellipse",
            },
          ],
          "type": "group",
        },
        "type": "symbol",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "bold",
          "text": "ellipse",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {},
        "mark": {
          "attrs": {},
          "children": [
            {
              "attrs": {
                "lineDash": null,
                "lineWidth": 1,
                "radius": 2,
                "stroke": "#3b4044",
              },
              "type": "circle",
            },
          ],
          "type": "group",
        },
        "type": "symbol",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "bold",
          "text": "circle",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {},
        "mark": {
          "attrs": {},
          "children": [
            {
              "attrs": {
                "lineDash": null,
                "lineWidth": 1,
                "radius": 2,
                "stroke": "#3b4044",
              },
              "type": "path",
            },
          ],
          "type": "group",
        },
        "type": "symbol",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "bold",
          "text": "diamond",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
      {
        "attrs": {
          "fill": "transparent",
          "lineDash": null,
          "lineWidth": 1,
          "radius": 2,
          "stroke": "transparent",
        },
        "type": "rect",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 14,
          "fontWeight": "bold",
          "text": "plaintext",
          "textAlign": "center",
          "textBaseline": "middle",
        },
        "type": "text",
      },
    ],
    "type": "group",
  },
}
`;
